### PR TITLE
Update ShippingMethod.php

### PR DESCRIPTION
**Pull request by marcomessa**
Add return type "mixed" to jsonSerialize, in order to make it compatible with Illuminate\Database\Eloquent\Model::jsonSerialize(): mixed

### CODE REVIEW

**Comment by tobias-kuendig** [comment]
Hm, this was just removed recently. We need to find a way to keep things compatible across Laravel/PHP versions.

https://github.com/OFFLINE-GmbH/oc-mall-plugin/commit/a9db1a879df5320a4732ed66316a66489fd41dd9

**Comment by marcomessa** [comment]
I'm sorry, I haven't noticed! Anyway, I can confirm that this is needed with Laravel 9 / October 3 / Php > 8, else the exception is thrown!

**Comment by mjauvin** [comment]
@tobias-kuendig @marcomessa I tried with the following with PHP 7.4/8.0/8.1 and it doesn't complain in either environment:
```
public function jsonSerialize(): array
```

**Comment by waynelogic** [comment]
OFFLINE! Please accept this pull request. I have to edit this line every time after installing the plugin.

File: models/ShippingMethod.php [hunk 1]
```diff
@@ -234,7 +234,7 @@ protected function priceRelation(
         return $this->priceAccessorPriceRelation($currency, $relation, $filter);
     }
 
-    public function jsonSerialize()
+    public function jsonSerialize(): mixed
```
Comment (mjauvin) [inline models/ShippingMethod.php:237]:
```suggestion
    public function jsonSerialize(): array
```

**Comment by tobias-kuendig** [comment]
Thank you @mjauvin!

**Comment by mjauvin** [comment]
Fixes #937

**Comment by marcomessa** [comment]
Thank you! I'm sorry I was off for a week and didn't see emails :D cheers!!

**Pull request merged**